### ISSUE TITLE
shell_integration no-title broken again

### ISSUE THREAD

**Issue by runeimp** [kovidgoyal/kitty#4955]
**Describe the bug**
`shell_integration no-title` in `kitty.conf` would allow you to set the window and tab to a specific string using ANSI escape codes. The latest update to `0.25.0` broke this behavior.

**To Reproduce**
1. Open KiTTY and in any window type `printf "\033]0;TITLE_TEST\007"`
2. This should set the window title and terminal tab to `TITLE_TEST`
3. This happens but is just a flash as it immediately switched back to the default behavior displaying the current path relative to `~` when applicable

**Screenshots**

**Environment details**
```
Key action failed

debug_config
invalid literal for int() with base 10: ''

Press Enter to close
```

**Additional context**
The expected title behavior does work but only with certain `shell_integration` configurations such as

* `shell_integration disabled`
* `shell_integration no-rc no-cursor no-title no-cwd no-prompt-mark no-complete`
* `shell_integration no-rc no-cursor no-title no-cwd no-prompt-mark`
* `shell_integration no-rc no-cursor no-title no-cwd`
* `shell_integration no-rc no-cursor no-title`
* `shell_integration no-rc no-title`

Oddly enough,  `shell_integration no-rc no-cursor` and any other single option other than `disabled` does not work, allowing you to set the title. This is on a MacBook Pro x86_64 running macOS Big Sur 11.6.5. I've not tested it on anything else.

**Comment by kovidgoyal** [comment]
Does not reproduce for me with

kitty --config=NONE -o 'shell_integration no-title'

and shell zsh.

If you are using fish, fish has its own title setting mechanism, which
kitty does not override, as noted in the docs: https://sw.kovidgoyal.net/kitty/shell-integration/

**Comment by runeimp** [comment]
This is macOS and I've never setup anything to mess with the window title. But I think you're missing the point. `shell_integration disabled` allows me to set the title. `shell_integration no-rc no-title` or every other combination of `no-title` and another option allows me to set the title. But `shell_integration no-title` doesn't allow me to set the title. Also this worked fine for the past 4 years or so until things changed recently (a few months ago?) and I needed to start using `shell_integration no-title` to allow me to set the title. Until yesterday with the latest update of KiTTY in Homebrew.

**Comment by kovidgoyal** [comment]
I'm afraid I cannot replicate, and I dont see how what you are
describing is possible.

**Comment by runeimp** [comment]
Yep, you guys were right. It was a issue in my `.bashrc` file. Once I removed

```
if test -n "$KITTY_INSTALLATION_DIR"; then
	export KITTY_SHELL_INTEGRATION="enabled"
	source "$KITTY_INSTALLATION_DIR/shell-integration/bash/kitty.bash"
fi
```

updating the title worked fine in KiTTY. If I comment out everything in `.bash_profile` except the source to `.bashrc` updating the title doesn't work. If I comment out everything in `.bashrc` except that block of code, which I've had in my `.bashrc` file for years, the title updating doesn't work. If I comment out nothing but that code the title updating works. What is going on?

**Comment by page-down** [comment]
> ... What is going on?

- `.bashrc` is loaded by `.bash_profile`.
- `export KITTY_SHELL_INTEGRATION="enabled"` is equivalent to overriding the `shell_integration` option in kitty.conf and enabling all.
- The shell integration sets the title by default, for example at $HOME the title will be `~`.

If you have configured it manually, you can disable shell integration in kitty.conf and then set `no-title` in bashrc.

**Comment by kovidgoyal** [comment]
Read https://sw.kovidgoyal.net/kitty/shell-integration especially the
part about manual shell integration to understand

**Comment by runeimp** [comment]
OK, was the code in `.bashrc` not _required_ in prior versions of KiTTY to enable shell integration?  If I'm correct, and prior versions of KiTTY did require that code to enable shell integration then this issue will affect a large percentage of users that bother to upgrade KiTTY on any kind of a regular basis. I wasted many hours attempting to figure out the problem prior to posting this issue. And then many more hours after that. Once again, if I'm correct that this was a required setup in a prior version of the app then this _IS_ a kitty problem because it blindsides it's users by simply upgrading the app.

And if I'm not correct that prior versions of the app didn't require shell configuration code it still wasn't a problem until this last update. Can't KiTTY detect this situation and issue a warning? Also shouldn't there be a warning about this potential situation in the docs instead of analyzing the docs to deduce the problem? Requiring the user to either RTFM with full analysis mode on or reinstall their OS without shell configuration of any kind to be able to upgrade without being blindsided is a pretty high bar. Considering how awesome KiTTY is (and I truly mean that) I'm very disappointed in this situation.

**Comment by page-down** [comment]
> ... If I'm correct, and prior versions of KiTTY did require that code to enable shell integration ...

Automatic shell integration does not require these.
The code automatically added in previous versions did not contain the `export KITTY_SHELL_INTEGRATION=...` part.

> ... Requiring the user to either RTFM with full analysis mode ... without being blindsided is a pretty high bar.

I think most people who read the manual configuration will be able to notice the first line.

```text
First, in kitty.conf set:
shell_integration disabled
```

and below:

```text
The value of `KITTY_SHELL_INTEGRATION` is the same as that for `shell_integration` ..
```

> `shell_integration no-title` in `kitty.conf` would allow you to ...
> ... The latest update to 0.25.0 broke this behavior.

If indeed, as you say, `kitty.conf` `no-title` still works after configuring `KITTY_SHELL_INTEGRATION=enabled`, then this can be considered a bug that has been fixed in the new version, and not a breaking change.

> ... I wasted many hours ...

I can understand your feelings, but unfortunately kitty still requires some understanding.
Feel free to ask questions in Discussions, I'm happy to answer if I can help.

**Comment by runeimp** [comment]
"kitty still requires some understanding"

I've been a programmer for over 2 decades in multiple languages. Sure, it's completely my fault, whatever. At least the problem is being resolved in the code per #4964. I'll take that as acknowledgement that this was in fact an issue.

Thank your civility. It is appreciated. But I'm having a hard time with all this talk about how unreasonable I seem to be with my experience with this update and observations about it. If I'm the only person in the world that suffered the consequences of this, great. With that fix going in so quickly hopefully we'll never know the actual impact, if any.

It is my greatest hope that all users of this app have the same experience as me, up until the last update. This has not been a problematic or hard to understand terminal app. At least not in the context of the field of terminal apps in general. Good luck.
